web/src/views/repo/build/BuildWrapper.vue

```diff
@@ -36,7 +36,11 @@
         <Tabs v-model="activeTab" disable-hash-mode class="order-2 md:order-none">
           <Tab id="tasks" :title="$t('repo.build.tasks')" />
           <Tab id="config" :title="$t('repo.build.config')" />
-          <Tab id="changed-files" :title="$t('repo.build.files', { files: build.changed_files?.length || 0 })" />
+          <Tab
+            id="changed-files"
+            v-if="!(build.event === 'tag' || build.event === 'cron' || build.event === 'deploy')"
```

Comment:
It would be more future proof if this would specify events that should show them instead (push, pull_request)